tests/monotouch-test/Network/NWBrowserTest.cs

```diff
@@ -102,17 +99,9 @@ public void TestStateChangesHandler ()
 						browserReady.Set ();
 				});
 				browser.SetChangesHandler ((oldResult, newResult) => {
-					// first time, listener appears, so we do not have an old result, second time
-					// listener goes, so we do not have a new result
+					didRan = true;
 					try {
-						if (firstRun) {
-							Assert.IsNull (oldResult, "oldResult first run.");
-							Assert.IsNotNull (newResult, "newResult first run");
-							firstRun = false;
-						} else {
-							Assert.IsNotNull (oldResult, "oldResult first run.");
-							Assert.IsNull (newResult, "newResult first run");
-						}
+						Assert.IsTrue (oldResult != null || newResult != null);
```

Comment:
That's not correct* The assert will crash the test app since it fail outside the main thread.
The result should be kept in a local variable and checked after the `WaitOne`

`*` I had a similar issue on the above `SetStateChangesHandler` a while ago (need to find where my diff is hiding).